packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart

```diff
@@ -537,9 +553,9 @@ class _ChatWidgetState extends State<ChatWidget> {
         Map<String, dynamic> location =
             functionCall.args['location']! as Map<String, dynamic>;
         var date = functionCall.args['date']! as String;
-        var latitude = location['latitude']! as double;
-        var longitude = location['longitude']! as double;
-        final functionResult = await fetchWeather(latitude, longitude, date);
+        var city = location['city']! as String;
+        var state = location['state']! as String;
```

Comment:
A nit from the existing pattern - an `as String` also checks against null (`as String?` would allow null) so the `!` is redundant. You might consider fixing the lines above too.

```suggestion
        var city = location['city'] as String;
        var state = location['state'] as String;
```